zib/StructureDefinitions/zib-AnatomicalLocation.xml

```diff
@@ -0,0 +1,92 @@
+  <derivation value="constraint" />
+  <differential>
+    <element id="CodeableConcept">
+      <path value="CodeableConcept" />
+      <short value="Location" />
+      <definition value="Localisation on/in the body." />
+      <comment value="In cases where only the location is required, or where laterality is embedded within the location code, this CodeableConcept can be used within the resource. However, when laterality needs to be explicitly recorded, the BodyStructure resource should be used, referenced through the BodyStructure Reference within this CodeableConcept. If the BodyStructure reference is used, this CodeableConcept should either remain empty or align with the `BodyStructure.location` element.&#xD;&#xA;&#xD;&#xA;**Note:** Receiving systems must support both this CodeableConcept and the BodyStructure resource, as either mechanism may be employed. The rationale for this is that, in many cases, only the location will be required, simplifying the FHIR recording process for sending systems." />
+      <alias value="Locatie" />
+      <mapping>
+        <identity value="zib-anatomicallocation-v1.0.3-2024EN" />
+        <map value="NL-CM:20.7.4" />
+        <comment value="Location" />
+      </mapping>
+    </element>
+    <element id="CodeableConcept.extension:bodyStructureReference">
+      <path value="CodeableConcept.extension" />
+      <sliceName value="bodyStructureReference" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://hl7.org/fhir/StructureDefinition/bodySite" />
+      </type>
+      <isModifier value="false" />
+    </element>
+    <element id="CodeableConcept.extension:bodyStructureReference.value[x]">
+      <path value="CodeableConcept.extension.value[x]" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-AnatomicalLocation-BodyStructure" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/BodyStructure" />
+      </type>
+    </element>
+    <element id="CodeableConcept.coding">
+      <path value="CodeableConcept.coding" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="$this" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+      <comment value="A deviation from our general profiling has been implemented in this profile. Slicing has been applied to `CodeableConcept.coding` to enable the use of the `BodyStructure` reference extension without requiring a populated coded value. This approach facilitates scenarios where only the Laterality element needs to be conveyed.&#xA;&#xA;The decision was made due to the Location concept’s required binding strength. If the required binding were applied on the CodeableConcept, it would mandate a coded value. This slicing approach ensures flexibility while maintaining compliance with the required binding constraints for Location." />
```

Comment:
Unless I'm missing something, I don't think this slicing trick is going to help. From the zib perspective, the desired solution is to
- either use (at least) a code from the zib value set
- or use a reference to BodyStructure

Since the slice is optional, the use of a code from this value set is also optional -- any other code would be valid. Effectively, this has the same effect as an _extensible_ binding on the data type itself.

I guess an invariant could be used to satisfy the requirement, but that might be a bit overkill.

Reply:
From the zib perspective the desired solution for the 'Location' concept is indeed to either have a code from the zib value sets here or in the BodyStructure resource via the reference. 

Normally in our FHIR profiling, if we bind a required value set, there is still the option to add additional codes (because Coding is 0..*).  This is still honored with this slicing trick.

The current slicing has indeed _almost_ the same effect as an extensible binding on the CodeableConcept. It saves adding the binding in the host profiles. ;) 

We could make an invariant that checks that if Codings are uses, alt least one Coding needs to have one of the systems from the ValueSet (http://snomed.info/sct and urn:oid:2.16.840.1.113883.6.43.1) which will help make sure it falls in the 0..1 slice for location? Not 100% watertight because a descendent-of is used for snomed...)

Reply:
@pieter-edelman-nictiz  added an invariant - tested it and it seems to work with .NET and Java QA stuff. What do you think?

Reply:
Nice work with the constraint!

But I'm not in favor of this slicing construct. It might make it easier for us but, but I think it makes it harder for profile consumers.

Reply:
@pieter-edelman-nictiz  I don't see too much of a problem with it for profile consumers.

The only other option we have is to add the ValueSet to the CodeableConcept in the host profile with an Extensible binding instead of Required? Or am I missing something here?

Reply:
This will result in a lot of entries in the known-issues.yml, to silence the zib-compliance checks. Correct?

Reply:
Ok! Next proposal is with the extensible binding. I made the changes. 
The known-issue.yml will need to be enriched in the host profiles that use this I think. We will see.